### PR TITLE
plan replayer: fix cannot load bindings when the statement contains in (...)

### LINKED ISSUE #43192: plan replayer cannot load bindings when the statement contains `in (...)`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table t1(a int);
create global binding for select * from t1 where a in (4, 5, 6) using select * from t1 where a in (4, 5, 6);
plan replayer dump explain select * from t1 where a in (4, 5, 6);
```

and then use the dumped replayer file to restore:

```
mysql> PLAN REPLAYER LOAD '/Users/wangchao/Downloads/rr/rr2.zip';
Query OK, 0 rows affected, 2 warnings (0.05 sec)

mysql> show warnings;
+---------+------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                                                                                                                                                                             |
+---------+------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Error   | 1064 | You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 70 near "... ) USING SELECT * FROM `test`.`t1` WHERE `a` IN (4,5,6)"                                         |
| Warning | 1105 | load bindings failed, err:[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 70 near "... ) USING SELECT * FROM `test`.`t1` WHERE `a` IN (4,5,6)"  |
+---------+------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
2 rows in set (0.00 sec)
```


### 2. What did you expect to see? (Required)

No warning, the bindings should be loaded

### 3. What did you see instead (Required)

Some warnings reported.

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by King-Dylan**
### What problem does this PR solve?

Issue Number: close #43192

Problem Summary:
plan replayer cannot load bindings when the statement contains in (...).

### What changed and how does it work?
This PR is to change the way of normalizing in-list from (…) to (?).

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/digester.go [hunk 1]
```diff
@@ -95,9 +95,9 @@ func Normalize(sql string) (result string) {
 // which removes general property of a statement but keeps specific property.
 //
 // for example: NormalizeForBinding('select 1 from b where a = 1') => 'select ? from b where a = ?'
-func NormalizeForBinding(sql string) (result string) {
+func NormalizeForBinding(sql string, isPlanReplayer bool) (result string) {
```
Comment (AilinKid) [inline pkg/parser/digester.go:98]: how about changing the var name as `forPlanReplayerReload`, which will be more clear here

Comment (AilinKid) [reply pkg/parser/digester.go:98]: the deep call chain name is the same

File: pkg/parser/digester.go [hunk 2]
```diff
@@ -230,10 +230,11 @@ func (d *sqlDigester) normalize(sql string, keepHint bool, forBinding bool) {
 		}
 
 		d.reduceLit(&currTok)
-
-		// Apply binding matching specific rules
-		if forBinding {
-			// IN (?) => IN ( ... ) #44298
+		if isPlanReplayer {
+			// Apply for plan replayer matching specific rules
```
Comment (AilinKid) [inline pkg/parser/digester.go:234]: pls make the comment more detailed, why for this code special for plan replayed reload binding and so on

File: pkg/server/handler/optimizor/plan_replayer_test.go [hunk 3]
```diff
@@ -248,6 +248,101 @@ func prepareData4PlanReplayer(t *testing.T, client *testserverclient.TestServerC
 	return filename, filename3
 }
 
+func TestIssue43192(t *testing.T) {
```
Comment (AilinKid) [inline pkg/server/handler/optimizor/plan_replayer_test.go:251]: test lgtm

**Review by AilinKid — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Comment by King-Dylan** [comment]
/needs-cherry-pick-release-7.5

**Pull request merged**